plugin/storage/es/es_mapping_generator.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package es
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/spf13/cobra"
+)
+
+// NewEsMappingsCommand defines the new CLI command to run esmapping-generator
+func NewEsMappingsCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "es-mappings",
+		Short: "Print Elasticsearch index mappings to stdout",
+		Run: func(_ *cobra.Command, _ []string) {
+			if _, err := os.Stat("./cmd/esmapping-generator"); os.IsNotExist(err) {
+				log.Fatalf("esmapping-generator not found: %v", err)
+			}
+
+			// Run the esmapping-generator command
+			execCmd := exec.Command("go", "run", "./cmd/esmapping-generator")
```

Comment:
How do you imagine this working in production if someone runs Jaeger as a container image? Where would they get `go` toolchain and Jaeger source code?

Reply:
Can you guide me here little

Reply:
Can you guide me here little

Reply:
first, I suggest you focus on one tool at a time, start with the cleaner only. The logic of the cleaner is already implemented in Go, we can call it directly as a library from the ES implementation. You can reuse all the sub-commands defined in cmd/es-cleaner, but you may have to move them out of main() into a reusable package (that would be a standalone refactoring and should be its own PR).

Reply:
one more thing - I think the easiest too to start with is es-mapping, because it's purely an on-demand command run with no need to set up periodic jobs.

I was wrong in the comment above - I don't think sub-commands will be needed once the cleaner/rollover are integrated into the main binary, because the only reason those sub-commands exist today is to allow the script to be run as externally managed cron job, so it needs to do different things. But if it's run as internal periodic job, it's one combined functionality.